### PR TITLE
feat: added db provider, removed init-config and default yaml file, updated tests

### PR DESCRIPTION
**Pull request by CL-SHLOMIKONCHA**
| Question                | Answer                                                                          |
| ---------------- | -------------------------------------------------------------------------- |
| Bug fix         | ✖                                                                        |
| New feature     | ✔                                                                      |
| Breaking change | ✔                                                                        |
| Deprecations    | ✖                                                                        |
| Documentation   | ✖                                                                        |
| Tests added     | ✔                                                                        |
| Chore            | ✔                                                                       |

### CODE REVIEW

File: src/common/interfaces.ts [hunk 1]
```diff
@@ -114,10 +134,14 @@ export interface IUpdateMosaicRequest {
 }
 
 export interface IFileProvider {
```
Comment (asafMasa) [inline src/common/interfaces.ts:136]: rename IFileProvider to IConfigAdapterProvider

File: src/containerConfig.ts [hunk 2]
```diff
@@ -4,25 +4,27 @@ import config from 'config';
...
   container.register(Services.LOGGER, { useValue: logger });
   container.register(Services.MAPPROXY, { useValue: mapproxyConfig });
+  container.register(Services.S3, { useValue: s3Config });
   container.register(Services.FILEPROVIDER, {
     useFactory: (): IFileProvider => {
-      return mapproxyConfig.fileProvider === Providers.S3 ? new S3Provider(container) : new FSProvider();
+      return getProvider(mapproxyConfig.fileProvider);
     },
   });
+  container.register(Services.PG, { useClass: PGClient });
```
Comment (asafMasa) [inline src/containerConfig.ts:27]: move this row to the other providers section (line 21)

File: src/getProvider.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+export const getProvider = (provider: string): IFileProvider => {
+  switch (provider.toLowerCase()) {
+    case 'fs':
+      return new FSProvider();
+    case 's3':
+      return new S3Provider();
+    case 'db':
+      return new DBProvider();
+    default:
+      return new FSProvider();
```
Comment (asafMasa) [inline src/getProvider.ts:16]: better to throw exception

File: src/common/middlewares/RollBackErrorHandler.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+@injectable()
+export class RollBackErrorHandler {
+  private readonly pgClient: IPGClient;
+  public constructor(@inject(Services.LOGGER) private readonly logger: ILogger, @inject(Services.MAPPROXY) private readonly config: IMapProxyConfig) {
+    this.pgClient = container.resolve<IPGClient>('PG');
+  }
+
+  public getRollBackHandlerMiddleware(): ErrorRequestHandler {
+    return async (err: Error, req: Request, res: Response, next: NextFunction): Promise<void> => {
+      if (this.config.fileProvider.toLowerCase() === Providers.DB) {
```
Comment (asafMasa) [inline src/common/middlewares/RollBackErrorHandler.ts:16]: is it relevant for "Begin" message?

**Review by asafMasa — CHANGES_REQUESTED** [review]
Good Job 👍

File: config/custom-environment-variables.json [hunk 5]
```diff
@@ -10,22 +10,33 @@
...
-      "awsAccessKeyId": "AWS_ACCESS_KEY_ID",
-      "awsSecretAccessKey": "AWS_SECRET_ACCESS_KEY",
-      "endpointUrl": "S3_ENDPOINT_URL",
-      "storageBucket": "S3_BUCKET",
-      "configFileBucket": "S3_CONFIG_FILE_BUCKET",
-      "objectKey": "S3_OBJECT_KEY",
-      "sslEnabled": "SSL_ENABLED"
+    }
+  },
+  "S3": {
```
[the remark this answers is not in the document]
Comment (CL-SHLOMIKONCHA) [reply config/custom-environment-variables.json:23]: fixed in 909f571

File: src/common/interfaces.ts (hunk 1 above)
Comment (CL-SHLOMIKONCHA) [reply src/common/interfaces.ts:139]: maybe IConfigProvider?

**Review by asafMasa — APPROVED** [review]

**Pull request merged**